pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarRecord.java

```diff
@@ -93,6 +95,15 @@ public Optional<Long> getEventTime() {
         }
     }
 
+    /**
+     * Some sink sometimes wants to control the ack type.
+     *
+     * @param cumulative
+     */
+    public void ack(boolean cumulative) {
```

Comment:
what about adding `ackCumulative` or `cumulativeAck` instead of a blindly method with a boolean param? 

it is also more similar to Consumer API

Reply:
I think this is similar to the API now provided, if provide `ackCumulative` or  `cumulativeAck` instead it, will easily make the original `ack` method understood as `Individual ack`.

Reply:
@nicoloboschi  Can you take a look at it again, thanks.

Reply:
I read again the pull. I believe that we should have
-  `cumulativeAck()` -> call cumulative acknowledgment
-  `individualAck()` -> call acknowledgment
- `ack()` -> based on the processing guaranteee

we need to explicity document that the first two will force the ack type, and the latter will use the "better" depending on the configuration

Reply:
Nice! I fixed. PTAL.